src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -606,7 +606,13 @@ bool primitive_inst::need_reset_output_memory() const {
         const bool is_user_onednn_impl = user_inst->get_node().get_preferred_impl_type() == impl_types::onednn;
         const bool is_user_conv = user_inst->get_node().is_type<convolution>();
         if (is_user_conv && is_user_onednn_impl) {
+            auto& conv_node = user_inst->get_node().as<convolution>();
             auto& output_layout = _impl_params->get_output_layout(0);
+            auto in_channel_count = get_convolution_channel_count(conv_node, output_layout, true);
+            // If the channel count is dynamic, we cannot verify feature alignment,
+            // so we conservatively skip the reset and return false for this condition.
+            if (in_channel_count == -1)
+                return false;
```

Comment:
I think the `conservative` behavior here is to reset memory for potential accuracy issue. Do we have a case where this makes difference in performance? If not, let's return true.

Reply:
Agree, return true is correct. Applied